pkg/standalone-utils/contracts/WeightedLPOracle.sol

```diff
@@ -0,0 +1,280 @@
+                _feedToken4DecimalScalingFactor = _calculateFeedTokenDecimalScalingFactor(feeds[i]);
+            } else if (i == 5) {
+                _feedToken5 = feeds[i];
+                _weight5 = weights[i];
+                _feedToken5DecimalScalingFactor = _calculateFeedTokenDecimalScalingFactor(feeds[i]);
+            } else if (i == 6) {
+                _feedToken6 = feeds[i];
+                _weight6 = weights[i];
+                _feedToken6DecimalScalingFactor = _calculateFeedTokenDecimalScalingFactor(feeds[i]);
+            } else if (i == 7) {
+                _feedToken7 = feeds[i];
+                _weight7 = weights[i];
+                _feedToken7DecimalScalingFactor = _calculateFeedTokenDecimalScalingFactor(feeds[i]);
+            }
+        }
+    }
+
+    /// @inheritdoc AggregatorV3Interface
+    function version() external view returns (uint256) {
+        return _version;
+    }
+
+    /**
+     * @notice Get the number of decimals present in the response value.
+     * @dev This is hard-coded to 18 decimals.
+     * @return decimals The number of decimals
+     */
+    function decimals() external pure returns (uint8) {
+        return 18;
+    }
+
+    /**
+     * @notice Get the description of the underlying aggregator that the proxy points to.
+     * @return description The description as a string
+     */
+    function description() external view returns (string memory) {
+        return _description;
+    }
+
+    /**
+     * @notice Get data about a specific round, using the roundId.
+     * @dev This is unused, and always returns all zeros.
+     * @return roundId The round ID
+     * @return answer The answer for this round
+     * @return startedAt Timestamp when the round started
+     * @return updatedAt Timestamp when the round was updated
+     * @return answeredInRound [Deprecated] - Previously used when answers could take multiple rounds to be computed
+     */
+    function getRoundData(
+        uint80 /* _roundId */
+    )
+        external
+        pure
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        return (0, 0, 0, 0, 0);
+    }
+
+    /**
+     * @notice Get the data from the latest round.
+     * @return roundId The round ID
+     * @return answer The answer for this round
+     * @return startedAt Timestamp when the round started
+     * @return updatedAt Timestamp when the round was updated
+     * @return answeredInRound [Deprecated] - Previously used when answers could take multiple rounds to be computed
+     */
+    function latestRoundData()
+        external
+        view
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        (int256[] memory prices, uint256 _updatedAt) = _getFeedData();
+
+        uint256 tvl = _calculateTVL(prices);
+        uint256 totalSupply = IERC20(address(pool)).totalSupply();
```

Comment:
Maybe as a small optimization, you can ask the vault directly what the total supply is (the pool is just a facade that calls the vault after all).